gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -220,6 +220,13 @@ private WriteObjectResponse doResumableUpload() throws IOException {
       // Only request committed size for the first insert request.
       if (writeOffset > 0) {
         writeOffset = getCommittedWriteSizeWithRetries(uploadId);
+        if (writeOffset % MAX_BYTES_PER_MESSAGE != 0) {
```

Comment:
Please add tests to cover the code path.

Could we get a confirmation from GCS team on the behaviour as well ?

Reply:
@veblush In one run of the benchmark we noticed this behavior. Towards the end of the file upload we received a transient error. On trying to resume, we fetched the committedWriteOffset in GCS. It returned a number which was not a multiple of 2MB(`MAX_BYTES_PER_MESSAGE`). Since we store uncommitted buffers in 2MB, we were not able to resume from that offset.

Can you confirm this?

Reply:
> Also, consider this scenario

In this case the complete file is written to the GCS. So when we retry, 
1. We get the file length as the committed Offset. 
2. We don't have a chunk beginning at that in our buffer, so 
3. We end up trying to read from the pipe, which returns EOF (0). 
4. This results in uploading `finalized` chunk to GCS with size as 0.